wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -503,6 +503,10 @@ internal class MapProtoAdapter<K, V> internal constructor(
     }
     reader.endMessageAndGetUnknownFields(token)
 
+    // Default to identity for the sake of scalars.
+    key = key ?: entryAdapter.keyAdapter.identity
+    value = value ?: entryAdapter.valueAdapter.identity
```

Comment:
2 things
I think we can set those default values when we create both `key` and `value` above.
We also wanna to do that **only for proto3**.

So that'd give us something like
```kotlin
    var (key: K?, value: V?) = when (syntax){
      Syntax.PROTO_2 -> null to null
      Syntax.PROTO_3 -> entryAdapter.keyAdapter.identity to entryAdapter.valueAdapter.identity
    }
```